### PR TITLE
Annotate EBPF API's with _Must_inspect_result_

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alan.jowett@microsoft.com>

Partially resolves: https://github.com/microsoft/ebpf-for-windows/issues/1542

## Description

Annotate EBPF APIs with _Must_inspect_result_.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

**Review by dthaler — APPROVED** [review]

**Comment by Alan-Jowett** [comment]
Blocked on https://github.com/microsoft/ebpf-for-windows/pull/1619

This change asserts on double close of file descriptors.

**Comment by Alan-Jowett** [comment]
Also blocked on: https://github.com/microsoft/ebpf-for-windows/pull/1623

**Review by dthaler — DISMISSED** [review]

**Review by saxena-anurag — APPROVED** [review]

File: tests/end_to_end/helpers.h [hunk 1]
```diff
@@ -161,13 +161,16 @@ typedef class _single_instance_hook : public _hook_helper
...
     {
-        ebpf_link_detach(link);
+        REQUIRE(ebpf_link_detach(link) == EBPF_SUCCESS);
     }
 
     void
     close_link(bpf_link* link)
     {
+#pragma warning(push)
+#pragma warning(disable : 6001) // Using uninitialized memory '*link'.
```
Comment (dthaler) [inline tests/end_to_end/helpers.h:171]: If you add SAL annotation (`_In_`) to line 168, will this warning go away?

File: libs/api/ebpf_api.cpp [hunk 2]
```diff
@@ -1039,25 +1039,33 @@ _link_ebpf_program(
...
+_clean_up_ebpf_link(_Frees_ptr_opt_ ebpf_link_t* link) noexcept
 {
     EBPF_LOG_ENTRY();
     if (link == nullptr) {
         EBPF_RETURN_VOID();
     }
+
+    // Suppress warning 6001: Using uninitialized memory '*link'.
+    // This is a false positive.  The memory is initialized in ebpf_link_create
+    // and ebpf_link_attach.
```
Comment (dthaler) [inline libs/api/ebpf_api.cpp:1058]: Was this warning there before removing `_In_opt_`?
If not, don't remove it/

**Review by dthaler — APPROVED** [review]
Approving and merging but see 2 questions remaining about compiler warnings

**Pull request merged**